ovs/extensions/generic/heartbeat.py

```diff
@@ -27,41 +27,64 @@
 class HeartBeat(object):
     """
     Heartbeat class
+    Put into crontab after installing openvstorage-core (see /etc/cron.d/openvstorage-core)
     """
     ARP_TIMEOUT = 30
 
-    def __init__(self):
-        """
-        Dummy init
-        """
-        raise Exception('Heartbeat class cannot be instantiated')
+    logger = Logger('extensions-generic')
 
-    @staticmethod
-    def pulse():
+    @classmethod
+    def pulse(cls):
         """
         Update the heartbeats for the Current Routers
         :return: None
         """
-        logger = Logger('extensions-generic')
-        machine_id = System.get_my_machine_id()
+        storagerouter = System.get_my_storagerouter()
         current_time = int(time.time())
+        with volatile_mutex('storagerouter_heartbeat_{0}'.format(storagerouter.guid)):
+            node_save = StorageRouter(storagerouter.guid)
+            node_save.heartbeats['process'] = current_time
+            node_save.save()
+        StorageRouterController.ping.s(storagerouter.guid, current_time).apply_async(routing_key='sr.{0}'.format(storagerouter.machine_id))
+
+        # Disabled because it no longer serves any purpose
+        # cls.clear_arp_cache(current_time)
+
+    @classmethod
+    def clear_arp_cache(cls, current_time):
+        """
+        Clear the ARP cache on this node for all other nodes in the cluster
+        :param current_time: Current timestamp to use
+        :type current_time: int
+        :return: None
+        :rtype: NoneType
+        """
+        for storagerouter in StorageRouterList.get_storagerouters():
+            if storagerouter == System.get_my_storagerouter():
+                continue
+            cls.clear_arp_cache_for_node(storagerouter, current_time)
+
+    @classmethod
+    def clear_arp_cache_for_node(cls, storagerouter, current_time):
+        """
+        Clear the ARP cache
+        Implemented for faster detections that a host is down
+        Unsure if relevant anymore. Here for documentation purposes
+        :param storagerouter: Node to clear ARP cache for
+        :type storagerouter: StorageRouter
+        :param current_time: Current timestamp to use
+        :type current_time: int
+        :return: None
+        :rtype: NoneType
+        """
+        current_time = current_time or time.time()
```

Comment:
As `current_time` is a mandatory param, time.time() will never be used. Either make it optional or remove this `or` statement